R/chop.R

```diff
@@ -90,188 +90,253 @@ chop <- function(data, cols) {
 #' @export
 #' @rdname chop
 unchop <- function(data, cols, keep_empty = FALSE, ptype = NULL) {
-  cols <- tidyselect::eval_select(enquo(cols), data)
-  if (length(cols) == 0) {
-    return(data)
-  }
+  sel <- tidyselect::eval_select(enquo(cols), data)
 
-  if (keep_empty) {
-    for (col in cols) {
-      data[[col]][] <- map(data[[col]], init_col)
-    }
+  if (is_null(ptype)) {
+    # Convert to form that `df_unchop()` requires
+    ptype <- list()
```

Comment:
As an internal detail, I've made `df_unchop()` always take a list `ptype`, so we convert that here